models/gif_psc_exp.cpp

```diff
@@ -0,0 +1,440 @@
+
+  ArrayDatum tau_sfa_list_ad( tau_sfa_ );
+  def< ArrayDatum >( d, names::tau_sfa, tau_sfa_list_ad );
+
+  ArrayDatum q_sfa_list_ad( q_sfa_ );
+  def< ArrayDatum >( d, names::q_sfa, q_sfa_list_ad );
+
+  ArrayDatum tau_stc_list_ad( tau_stc_ );
+  def< ArrayDatum >( d, names::tau_stc, tau_stc_list_ad );
+
+  ArrayDatum q_stc_list_ad( q_stc_ );
+  def< ArrayDatum >( d, names::q_stc, q_stc_list_ad );
+}
+
+void
+nest::gif_psc_exp::Parameters_::set( const DictionaryDatum& d )
+{
+
+  updateValue< double >( d, names::I_e, I_e_ );
+  updateValue< double >( d, names::E_L, E_L_ );
+  updateValue< double >( d, names::g_L, g_L_ );
+  updateValue< double >( d, names::C_m, c_m_ );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+  updateValue< double >( d, names::Delta_V, Delta_V_ );
+  updateValue< double >( d, names::V_T_star, V_T_star_ );
+
+  if ( updateValue< double >( d, names::lambda_0, lambda_0_ ) )
+  {
+    lambda_0_ /= 1000.0; // convert to 1/ms
+  }
+
+  updateValue< double >( d, names::t_ref, t_ref_ );
+  updateValue< double >( d, names::tau_syn_ex, tau_ex_ );
+  updateValue< double >( d, names::tau_syn_in, tau_in_ );
+
+  updateValue< std::vector< double > >( d, names::tau_sfa, tau_sfa_ );
+  updateValue< std::vector< double > >( d, names::q_sfa, q_sfa_ );
+  updateValue< std::vector< double > >( d, names::tau_stc, tau_stc_ );
+  updateValue< std::vector< double > >( d, names::q_stc, q_stc_ );
+
+  if ( tau_sfa_.size() != q_sfa_.size() )
+    throw BadProperty( String::compose(
+      "'tau_sfa' and 'q_sfa' need to have the same dimensions.\nSize of "
+      "tau_sfa: %1\nSize of q_sfa: %2",
+      tau_sfa_.size(),
+      q_sfa_.size() ) );
+
+  if ( tau_stc_.size() != q_stc_.size() )
+    throw BadProperty( String::compose(
+      "'tau_stc' and 'q_stc' need to have the same dimensions.\nSize of "
+      "tau_stc: %1\nSize of q_stc: %2",
+      tau_stc_.size(),
+      q_stc_.size() ) );
+
+  if ( g_L_ <= 0 )
+    throw BadProperty( "Membrane conductance must be strictly positive." );
+
+  if ( Delta_V_ <= 0 )
+    throw BadProperty( "Delta_V must be strictly positive." );
+
+  if ( c_m_ <= 0 )
+    throw BadProperty( "Capacitance must be strictly positive." );
+
+  if ( t_ref_ < 0 )
+    throw BadProperty( "Refractory time must not be negative." );
+
+  if ( lambda_0_ < 0 )
+    throw BadProperty( "lambda_0 must not be negative." );
+
+  for ( size_t i = 0; i < tau_sfa_.size(); i++ )
+    if ( tau_sfa_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  for ( size_t i = 0; i < tau_stc_.size(); i++ )
+    if ( tau_stc_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  if ( tau_ex_ <= 0 || tau_in_ <= 0 )
+    throw BadProperty( "Synapse time constants must be strictly positive." );
+}
+
+void
+nest::gif_psc_exp::State_::get( DictionaryDatum& d, const Parameters_& p ) const
+{
+  def< double >( d, names::V_m, y3_ );    // Membrane potential
```

Comment:
I thought [from the header file](https://github.com/nest/nest-simulator/pull/261/files#diff-0ae2dd2fbce391d89f0e0c9554c0f837R241) that ``y3_`` was supposed to be the relative potential, not the absolute ``V_m``...